### PR TITLE
8) added new models/services/dao

### PR DESCRIPTION
**Pull request by AntonPanchishko**

### CODE REVIEW

**Review by less08 — APPROVED** [review]

File: src/main/java/com/dev/cinema/Main.java [hunk 1]
```diff
@@ -1,18 +1,43 @@
...
 import com.dev.cinema.service.MovieService;
+import com.dev.cinema.service.MovieSessionService;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
 
 public class Main {
     private static Injector injector = Injector.getInstance("com.dev.cinema");
 
     public static void main(String[] args) {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: redundant empty line

File: src/main/java/com/dev/cinema/Main.java [hunk 2]
```diff
@@ -1,18 +1,43 @@
...
+        System.out.println(cinemaHallService.getAll());
+        MovieSession movieSession = new MovieSession();
+        movieSession.setMovie(movie);
+        movieSession.setCinemaHall(cinemaHall);
+        movieSession.setSessionTime(LocalDateTime
+                .of(2021, 2, 2, 10, 30));
+        MovieSessionService movieSessionService =
+                (MovieSessionService) injector.getInstance(MovieSessionService.class);
+        movieSessionService.add(movieSession);
+        movieSessionService.findAvailableSessions(1L, LocalDate
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: let's use `movie.getId()` instead

File: src/main/java/com/dev/cinema/model/CinemaHall.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package com.dev.cinema.model;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class CinemaHall {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/CinemaHall.java:11]: let's add the `@Table` annotation here with the `proper_table_name`

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import java.time.LocalDateTime;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+
+@Entity
+public class MovieSession {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/MovieSession.java:14]: same here

File: src/main/java/com/dev/cinema/model/CinemaHall.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private int capacity;
+    private String description;
+
+    public CinemaHall() {
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/CinemaHall.java]: do we need to create this?

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private Movie movie;
+    @ManyToOne
+    private CinemaHall cinemaHall;
+    private LocalDateTime sessionTime;
+
+    public MovieSession() {
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/MovieSession.java]: same

File: src/main/java/com/dev/cinema/model/CinemaHall.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    public int getCapacity() {
+        return capacity;
+    }
+
+    public void setCapacity(int capacity) {
+        this.capacity = capacity;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/CinemaHall.java:24]: let's create getters and setters for all the fields

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+    public Movie getMovie() {
+        return movie;
+    }
+
+    public void setMovie(Movie movie) {
+        this.movie = movie;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/MovieSession.java:32]: same

File: src/main/java/com/dev/cinema/Main.java [hunk 9]
```diff
@@ -1,18 +1,43 @@
...
+                .of(2021, 2, 2)).forEach(System.out::println);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java:40]: what gets printed out? can you please share the screenshot?

File: src/main/java/com/dev/cinema/model/MovieSession.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+@Entity
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private Movie movie;
+    @ManyToOne
+    private CinemaHall cinemaHall;
+    private LocalDateTime sessionTime;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/MovieSession.java:24]: let's use `@JoinColumn` to set `proper_column_names` in our table

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/Main.java (hunk 9 above)
Comment (AntonPanchishko) [reply src/main/java/com/dev/cinema/Main.java:40]: image

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**